lib/google/apis/core/upload.rb

```diff
@@ -263,6 +286,23 @@ def execute_once(client, &block)
           @last_error = e
           error(e, rethrow: true, &block)
         end
+
+        # @private
+        # Calculate a chunk_size as close to max_chunk_size as possible.
+        def calc_chunk_size granularity
+          chunk_multi = max_chunk_size / granularity
+          chunk_multi = 1 if chunk_multi.zero?
+          granularity * chunk_multi
+        rescue
+          nil
+        end
+
+        # @private
+        # The maximum chunk_size to use for resumable chunked uploads.
+        def max_chunk_size
+          # 1GB, magic value
+          1000000000
```

Comment:
As I think more about this, it seems that the chunk size will also dictate memory usage. We're reading this much content from the IO into a byte array, and then posting it (which, I don't know how httpclient works internally, but it's possible it would need to make another copy of the buffer). Given that, it seems a smaller chunk size may be in order. And indeed that it should be configurable for lower memory environments.

Reply:
Oh, I don't think I could have been more wrong. It looks like the [old default](https://github.com/googleapis/google-cloud-ruby/pull/703/files#diff-9faea63259273e1bfe612b10cbd7791bR93) was 10MB! What do you think about splitting the difference and changing this to 100MB?

Reply:
I'm actually inclined to go with 10MB (10_485_760) then.

Reply:
Hmm. Okay, I guess I'm okay with 100MB, especially if Python is using that value as well. I'm just really nervous about the memory requirement.